src/Core/Domain/ValueObject/Email.php

```diff
@@ -78,19 +78,16 @@ public function isEqualTo(Email $email)
     }
 
     /**
-     * Assert that email is in valid format
+     * Check that email is not an empty string
      *
-     * @param string $email
+     * @param $email
      *
      * @throws DomainConstraintException
      */
-    private function assertEmailIsValid($email)
+    public function assertEmailIsNotEmpty($email)
     {
-        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
```

Comment:
@atomiix Can you tell us why `filter_var($email, FILTER_VALIDATE_EMAIL)` is not right 😮 ?! it's both surprising and frightening somehow

Reply:
`FILTER_VALIDATE_EMAIL` is not rfc 6531 compliant. It doesn't allow email address like `prestashop@localhost` for instance. Also, as the real email validation is done by the `Validation::isEmail()` method, we should not be validating the email in another place but check for coherence in the ValueObject.

Reply:
Can we add this explanation into the code as a comment ? I am sure other developers will have the same confusion that I had, and in order to prevent them from thinking "omg prestashop is bad 🙄 " it would be nice to provide them the full context